src/solvers/smt2_incremental/smt2_incremental_decision_procedure.h

```diff
@@ -0,0 +1,36 @@
+
+#ifndef CBMC_SMT2_INCREMENTAL_DECISION_PROCEDURE_H
+#define CBMC_SMT2_INCREMENTAL_DECISION_PROCEDURE_H
+
+#include <solvers/stack_decision_procedure.h>
+
+class smt2_incremental_decision_proceduret final
+  : public stack_decision_proceduret
```

Comment:
No. We are planning to use the existing implementation in `solvers/smt2/smt2_conv.cpp` for reference, but not to reuse it verbatim. This will enable structuring the implementation differently. The existing implementation has: 

-  5000 lines of code within a single file.
-  Result parsing and formula generation in the same file/class.
-  Functions which convert directly from `exprt` to smt2 formulas as strings.
-  No unit tests.

We were concerned that adding a major new feature into the code for the existing implementation would be highly error prone and make it tricky to track down bugs when they do occur. In particular we would like to add separated:
 * Abstract representation of smt2 formulas.
 * Conversion from `exprt` to abstract smt2 formulas.
 * Conversion from abstract smt2 formulas to string.
 * Unit tests for the separate conversion steps.
The idea is that the above outlined structure would make the testing and debugging comparatively straight forward. We will learn as much as possible from the existing implementation in order make good use of the work which has been done previously.

Reply:
Seriously; look at SMT-Switch.  That already has most of what you need.

Reply:
Using `SMT-Switch` is an interesting idea and it warrants further discussion within Diffblue. So please do not think that I am discounting that option. However I can see at least 2 potential issues with that approach, namely:

1. The `SMT-switch` repository you linked to has no Windows CI. CBMC does have Visual Studio / Windows CI, testing and packages. So it would be good to choose a solution which allows us to keep parity between Windows support and support for Unix-like operating systems.
2. There is a desire for it to be possible to swap in new / different solvers at run time. Is that something which can be supported through `SMT-switch`, or would we be stuck with supporting only fixed versions of selected solvers linked to CBMC at build / compile time?

Reply:
This seems like a conversation for the issue RFC - https://github.com/diffblue/cbmc/issues/6134
Ok, if I mark this as resolved for the minute as it doesn't really relate to the work which is currently in this PR?

Reply:
Sure!